### PR TITLE
[CI] Update pre-commit and remove ros-lint

### PR DESCRIPTION
**Pull request by christophfroehlich**
Remove ros-lint CI job but change configuration of pre-commit to run the linters with the ci-format job (renamed to ci-pre-commit) as discussed [here](https://github.com/ros-controls/ros2_control_ci/pull/11#issuecomment-1951455803).

Do we need the matrix strategy here, or is it sufficient to check on rolling distro?

Format fails until #188 gets merged

### CODE REVIEW

File: .github/workflows/build-binary.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ on:
 
 jobs:
   binary:
-    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-industrial-ci-with-cache.yml@master
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-industrial-ci-with-cache.yml@ci_lint
```
Comment (christophfroehlich) [inline .github/workflows/build-binary.yml:17]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-industrial-ci-with-cache.yml@master
```

File: .github/workflows/build-source.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@ on:
 
 jobs:
   source_build:
-    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-ros-tooling-source-build.yml@master
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-ros-tooling-source-build.yml@ci_lint
```
Comment (christophfroehlich) [inline .github/workflows/build-source.yml:13]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-ros-tooling-source-build.yml@master
```

File: .github/workflows/ci-ros-lint.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: ROS 2 Lint
+on:
+  pull_request:
+
+jobs:
+  lint_rolling:
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-ci-ros-lint.yml@ci_lint
```
Comment (christophfroehlich) [inline .github/workflows/ci-ros-lint.yml:7]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-ci-ros-lint.yml@master
```

File: .github/workflows/ci-ros-lint.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    with:
+      ros_distro: rolling
+  lint:
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-ci-ros-lint.yml@ci_lint
```
Comment (christophfroehlich) [inline .github/workflows/ci-ros-lint.yml:11]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-ci-ros-lint.yml@master
```

File: .github/workflows/debian-build.yml [hunk 5]
```diff
@@ -11,7 +11,7 @@ on:
 
 jobs:
   debian_source_build:
-    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-debian-build.yml@master
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-debian-build.yml@ci_lint
```
Comment (christophfroehlich) [inline .github/workflows/debian-build.yml:14]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-debian-build.yml@master
```

File: .github/workflows/rhel-semi-binary-build.yml [hunk 6]
```diff
@@ -11,7 +11,7 @@ on:
 
 jobs:
   rhel_semi_binary_build:
-    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-rhel-binary-build.yml@master
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-rhel-binary-build.yml@ci_lint
```
Comment (christophfroehlich) [inline .github/workflows/rhel-semi-binary-build.yml:14]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-rhel-binary-build.yml@master
```

File: .github/workflows/ci-pre-commit.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+name: Format
+
+on:
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  pre-commit:
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-pre-commit.yml@ci_format
```
Comment (christophfroehlich) [inline .github/workflows/ci-pre-commit.yml:9]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-pre-commit.yml@master
```

File: .github/workflows/ci-pre-commit.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
```
Comment (christophfroehlich) [inline .github/workflows/ci-pre-commit.yml:9]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-pre-commit.yml@master
```

File: .github/workflows/ci-pre-commit.yml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+name: Format
```
Comment (fmauch) [inline .github/workflows/ci-pre-commit.yml:1]:
```suggestion
name: Pre-Commit
```

Or is there a reason you didn't change it?

**Review by fmauch — APPROVED** [review]
Looks good to me :-)

**Review by bmagyar — APPROVED** [review]

**Pull request merged**